Better support for C enum.

stephan57160 opened the pull request:
See discussion in #419.

Create a 'enum' in .h file, with same values than in C++.
Difference are the names of the 'constants' : In C, the enum name is prepended before the RUST enum name.

Testing tool are also available in tests/expectations.

Note:
- `generate_c_code_for_enum` is renamed as `generate_cpp_code_for_enum`, as it is actually generating code for C++.
- `generated_c_code_for_enum` is created to generate C code !

Dushistov commented on macroslib/src/cpp/fenum.rs:
```diff
@@ -142,6 +144,71 @@ enum {enum_name} {{"#,
+        file,
+        r"}};
+
+typedef enum {enum_name} {enum_name}Opaque;
```
All looks good, except of naming (the most complex question of programming). 
"Opaque" elsewhere in crate used for some kind of [type erasure](https://en.wikipedia.org/wiki/Type_erasure). 
So "opaque" is related to that real type is invisible from `<Type>Opaque`. In case of `enum` type is visible,
so I suggest name it in some other way. For example "C{enum_name}`

stephan57160 replied:
Fixed.

Dushistov commented on macroslib/tests/expectations/cenum.cpp:
```diff
@@ -0,0 +1,21 @@
+// Important !
+// This new type needs to be named as 'SomeEnumOpaque', 
+// else this conflicts with C++ bindings code (type 'SomeEnum' is already defined there).
+r#"enum SomeEnum {
```
In this case more robust test case would be to add the whole file content to test with "pragma once and so on",
to distinguish from C++ header. Because of this test passes even without  new `generate_c_code_for_enum`,
because of C++ header contains the same code.

stephan57160 replied:
Fixed.

stephan57160 commented:
I see also a conflicting files, which was nos there previsously.
I'll change that too, no problem.

You prefer different commits, or a squash of all ?

Dushistov commented:
> You prefer different commits, or a squash of all ?

I suppose "squash" is better for such small feature, too make life easier during "git bisect".

> I see also a conflicting files

I removed empty line in enum generator, and I suppose this is cause conflicts.

stephan57160 commented:
> > You prefer different commits, or a squash of all ?
> 
> I suppose "squash" is better for such small feature, too make life easier during "git bisect".
> 
> > I see also a conflicting files
> 
> I removed empty line in enum generator, and I suppose this is cause conflicts.

No. 
Mistake was on my side. 
I simply did not fetch upstream/master and did not get the new 'enum syntax' PR.
Before I understood this, I had a few push attempts ...
It's fixed now, and I added some more tests, with this too.
